torch/utils/inference/run.py

```diff
@@ -0,0 +1,569 @@
+r"""
+This is a script for launching PyTorch inference on Intel(R) Xeon(R) Scalable Processors with optimal configurations.
+Single instance inference, multi-instance inference are enabled.
+
+To get the peak performance on Intel(R) Xeon(R) Scalable Processors, the script optimizes the configuration of thread and memory
+management. For thread management, the script configures thread affinity and the preload of Intel OMP library.
+For memory management, it configures NUMA binding and preload optimized memory allocation library (e.g. tcmalloc, jemalloc).
+
+**How to use this module:**
+
+*** Single instance inference ***
+
+1. Run single-instance inference on a single node with all CPU nodes.
+
+::
+
+   >>> torchrun --inference --throughput_mode script.py args
+
+2. Run single-instance inference on a single CPU node.
+
+::
+
+   >>> torchrun --inference --node_id 1 script.py args
+
+*** Multi-instance inference ***
+
+1. Multi-instance
+   By default, one instance per node. if you want to set the instance numbers and core per instance,
+   --nprocs and  --ncore_per_instance should be set.
+
+::
+
+   >>> torchrun --inference -- python_script args
+
+   eg: on an Intel(R) Xeon(R) Scalable Processor with 14 instance, 4 cores per instance
+
+::
+
+   >>> torchrun --inference  --nprocs 14 --ncore_per_instance 4 python_script args
+
+2. Run single-instance inference among multiple instances.
+   By default, runs all nprocs. If you want to independently run a single instance among nprocs, specify instance_idx.
+
+   eg: run 0th instance on an Intel(R) Xeon(R) Scalable Processor with 2 instance (i.e., numactl -C 0-27)
+
+::
+
+   >>> torchrun --inference  --nprocs 2 --instance_idx 0 python_script args
```

Comment:
we typically use `--rank` for this, any reason why this is called `--instance_idx`? In general I'd like to be consistent with terminology since there is already a lot going on under the hood.

Reply:
Sure.